components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIManager.java

```diff
@@ -453,9 +453,34 @@ String addResourceFile(Identifier identifier, String resourcePath, ResourceFile
      * @param tenantid   tenant id
      * @return true if the scope key is already available
      * @throws APIManagementException if failed to check the context availability
+     * @deprecated Use isScopeKeyAssignedLocally() instead.
      */
+    @Deprecated
     boolean isScopeKeyAssigned(APIIdentifier identifier, String scopeKey, int tenantid) throws APIManagementException;
```

Comment:
can we remove it

Reply:
This was added back as it seemed not a good practice to remove an API https://github.com/wso2/carbon-apimgt/pull/8409#discussion_r409378751

Reply:
As the method is not used by any implementation other than internal carbon-apimgt methods, removed the method from interface. https://github.com/wso2/carbon-apimgt/pull/8476/commits/99c8da528391eede8b3d4adfab83a003c5c4ce7b